src/toil/__init__.py

```diff
@@ -145,10 +147,116 @@ def lookupEnvVar(name, envName, defaultValue):
         return value
 
 
+def checkDockerImageExists(appliance):
+    """
+    Attempts to check a url registry_name for the existence of a docker image with a given tag.
+
+    :param str appliance: The url of a docker image's registry (with a tag) of the form:
+                          'quay.io/<repo_path>:<tag>' or '<repo_path>:<tag>'.
+                          Examples: 'quay.io/ucsc_cgl/toil:latest', 'ubuntu:latest', or
+                          'broadinstitute/genomes-in-the-cloud:2.0.0'.
+    :return: Raises an exception if the docker image cannot be found or is invalid.  Otherwise, it
+             will return the appliance string.
+    :rtype: str
+    """
+    appliance = appliance.lower()
+    tag = appliance.split(':')[-1]
+    registry_name = appliance[:-(len(':' + tag))] # remove only the tag
+
+    # jenkins tests
+    if ('quay.io/ucsc_cgl/toil-pr' in registry_name) or ('quay.io/ucsc_cgl/toil-it' in registry_name):
```

Comment:
Why don't the tests use this mechanism?

Reply:
I don't catch your meaning.  Jenkins sets `TOIL_APPLIANCE_SELF` to `quay.io/ucsc_cgl/toil-pr` and `quay.io/ucsc_cgl/toil-it`, which are both invalid, on purpose.  Since this checks for valid docker images, it needs to skip these or jenkins will have a conniption.